src/models/geograph.py

```diff
@@ -321,51 +348,36 @@ def _load_from_dataframe(
         Args:
             df (gpd.GeoDataFrame): GeoDataFrame containing polygon objects from
             a shape file.
-            attributes (Optional[List[str]], optional): columns of df that are
-            added to nodes of graph as attributes. Defaults to None,
-            which will cause all attributes to be added.
             tolerance (float, optional): Adds edges between neighbours that are
             at most `tolerance` units apart. Defaults to 0.
 
         Raises:
-            ValueError: If `tolerance` < 0, if `class_label` is not a column in
-            the dataframe, or if `attributes` contains column names not in the
-            dataframe.
+            ValueError: If `tolerance` < 0, if `class_label` or `geometry` are
+            not columns in the dataframe.
 
         Returns:
-            STRtree: The Rtree object used to build the graph.
+            Tuple: A tuple with the Rtree object used to build the graph and the
+            dataframe containing polygon objects.
         """
         if tolerance < 0.0:
             raise ValueError("`tolerance` must be greater than 0.")
-        if (
-            attributes is not None
-            and "class_label" not in attributes
-            or "class_label" not in df.columns
-        ):
+        if "class_label" not in df.columns:
             raise ValueError("`class_label` must be a column in the dataframe.")
-        if (
-            attributes is not None
-            and "geometry" not in attributes
-            or "geometry" not in df.columns
-        ):
+        if "geometry" not in df.columns:
             raise ValueError("`geometry` must be a column in the dataframe.")
-        # If no attribute list is given, all
-        # columns in df are used
-        if attributes is None:
-            attributes = df.columns.tolist()
-        elif not set(attributes).issubset(df.columns):
-            raise ValueError("`attributes` must only contain column names in `df`.")
 
+        # Reset index to ensure consistent indices
+        df = df.reset_index(drop=True)
```

Comment:
By default `.reset_index` creates a copy of the original dataframe right? (So we're not running at risk of modifying variables outside of the function scope)

Reply:
Yeah, all dataframe functions usually have an `inplace` option which is `False` by default but is generally a bad idea to set to `True` https://stackoverflow.com/questions/45570984/in-pandas-is-inplace-true-considered-harmful-or-not